integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/statistics/PrometheusStatisticsTest.java

```diff
@@ -40,6 +42,8 @@ public void init() throws Exception {
     @Test(groups = { "wso2.esb" }, description = "Test if metric data are exposed, when it is enabled in config")
     public void testPrometheusPublisher() throws Exception {
 
+        Awaitility.await().pollInterval(50, TimeUnit.MILLISECONDS).atMost(60, TimeUnit.SECONDS).
+                until(isManagementApiAvailable());
```

Comment:
Prometheus has a different API than Management. I added a fix for it in https://github.com/wso2/micro-integrator/pull/577

Reply:
It is the internal API right?
In the logs this is what you get
INFO  [org.wso2.carbon.automation.extensions.servers.utils.ServerLogReader] - [2019-07-03 18:20:25,091] [micro-integrator]  INFO - PassThroughListeningIOReactorManager Pass-through EI_INTERNAL_HTTP_INBOUND_ENDPOINT Listener started on 0.0.0.0:9391
INFO  [org.wso2.carbon.automation.extensions.servers.utils.ServerLogReader] - [2019-07-03 18:20:25,128] [micro-integrator]  INFO - PassThroughListeningIOReactorManager Pass-through EI_INTERNAL_HTTPS_INBOUND_ENDPOINT Listener started on 0.0.0.0:9354

If the HTTPS port there, the HTTP port is also available

Reply:
Yes , because we start http and then https . In this way also it will work . I added a check to check http port , as that is what we really want in this check .